### DISCUSSION TITLE
How to set `ENV` based on cmd output in `install`

### DISCUSSION THREAD

**Discussion by themissingcow** [Homebrew/discussions#4887] [Writing Formulae/Casks] [answered]
### Output of `brew config`

```shell
homebrew-core (master) % brew config
HOMEBREW_VERSION: 4.1.19-21-g3baec05
ORIGIN: https://github.com/Homebrew/brew
HEAD: 3baec056e78708da77753d0e9b186938cb83b90a
Last commit: 14 hours ago
Core tap origin: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 1c3ef9056761ff586002f13dd6b927bab4808d60
Core tap last commit: 25 minutes ago
Core tap branch: master
Core tap JSON: 08 Nov 10:04 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_EDITOR: nvim
HOMEBREW_MAKE_JOBS: 12
HOMEBREW_SORBET_RUNTIME: set
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: dodeca-core 64-bit arm_blizzard_avalanche
Clang: 15.0.0 build 1500
Git: 2.39.3 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Curl: 8.1.2 => /usr/bin/curl
macOS: 14.1-arm64
CLT: 15.0.0.0.1.1694021235
Xcode: 15.0.1
Rosetta 2: false
```

### Output of `brew doctor`

```shell
homebrew-core (master) % brew doctor
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Unbrewed dylibs were found in /usr/local/lib.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected dylibs:
  /usr/local/lib/libtidy.5.8.0.dylib

Warning: Unbrewed static libraries were found in /usr/local/lib.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected static libraries:
  /usr/local/lib/libtidy.a

Warning: You have uncommitted modifications to Homebrew/homebrew-core.
If this is a surprise to you, then you should stash these modifications.
Stashing returns Homebrew to a pristine state but can be undone
should you later need to do so for some reason.
  cd /opt/homebrew/Library/Taps/homebrew/homebrew-core && git stash -u && git clean -d -f

Uncommitted files:
  ?? .faircamp_cache/CACHE_VERSION_MARKER_4
  ?? Formula/f/faircamp.rb
```

### Description of issue

Hey folks,

I need to write a formulae to build a cargo project that depends on `libvips`. This only ships with a dynamic library, and so needs exposing to the linker at build time.  For `<reasons>` (I'm not a Rust person), the correct linker flags aren't being set through Cargo.

For a normal build, this can be easily remedied by:

```
RUSTFLAGS=`pkg-config --libs vips` cargo install --features libvips --locked --path .
```

I'm struggling to work out how to setup the extra flags in a Formulae. I  couldn't see a way to get the output of `pkgconfig` into a variable for  `ENV["RUSTFLAGS"] = flags`, and trying the inline shell expansion shown above also fails (which makes sense). 

```
Failed to execute: RUST_FLAGS=`pkg-config --libs vips`
```

Sorry I'm probably missing something, I've searched in other formulae, but does anyone know how to get the output of a command in `install` in the same way as `shell_output` in `test?  ...or the "right" way to do this?

 I don't want to hard-code the flags. In this case, all the build dependencies, including `libvips` are being installed by `brew` and so are version dependent Cellar paths. 

Many thanks in advance,
Tom

**Comment by alebcay** [comment, marked as answer]
Hi Tom, you can use backticks to capture command output (you may be familiar with this in shell, but this is also a [Ruby-ism](https://stackoverflow.com/a/1563625)):

```ruby
ENV["RUSTFLAGS"] = `pkg-config --libs vips`.chomp # sets RUSTFLAGS in the environment
system "cargo", "install", "--features", "libvips", "--locked", "--path", "." # will use RUSTFLAGS set above
```

One similar example of this in existing formula is https://github.com/Homebrew/homebrew-core/blob/a1640c6c17a2f4bc14336d364f1989da8631e1c8/Formula/x/xsd.rb#L52.

You can generally use some other techniques for capturing command output in Ruby, including [`::capture2`, `::capture2e`, and/or `::capture3` in `Open3`](https://ruby-doc.org/stdlib-2.6.3/libdoc/open3/rdoc/Open3.html).

**Reply by themissingcow** [reply]
Thanks so much @alebcay - thats incredibly straight forward! Sorry I'm new to Ruby...
